### PR TITLE
chore: remove pinned dependencies

### PR DESCRIPTION
**Pull request by mdonnalley**
- Removes `pinnedDependencies` 
- Removes `verify-no-schema-changes` cronjob since it won't be able to test anything without `pinnedDependencies`
- Updates dev-scripts to 2.0.0

@W-10379933@

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -101,34 +101,12 @@
...
-    "@sf/login",
-    "@salesforce/plugin-telemetry",
-    "@oclif/plugin-autocomplete",
-    "@oclif/plugin-help",
-    "@oclif/plugin-not-found",
-    "@oclif/plugin-plugins",
-    "@oclif/plugin-update",
-    "@oclif/plugin-version"
-  ],
-  "resolutions": {
```
Comment (peternhale) [inline package.json:121]: @mdonnalley shouldn't these remain?

**Review by peternhale — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (mdonnalley) [reply package.json:121]: @peternhale maybe - I just wanted to see if the build passed without them

File: package.json [hunk 2]
```diff
@@ -142,33 +120,27 @@
     "postpack": "rm -f oclif.manifest.json",
     "posttest": "yarn test:deprecation-policy",
     "prepack": "sf-prepack",
-    "prepare": "yarn compile && yarn lint",
+    "prepare": "sf-install",
```
Comment (RodEsp) [inline package.json:123]: Do these do the same thing?

Comment (RodEsp) [reply package.json:123]: It doesn't seem like it based on [sf-install.js](https://github.com/forcedotcom/dev-scripts/blob/main/bin/sf-install.js). Why do we want to change this behavior?

Comment (mdonnalley) [reply package.json]: Ah I guess not

**Review by RodEsp — APPROVED** [review]

**Review by peternhale — APPROVED** [review]

**Review by peternhale — APPROVED** [review]

**Review by RodEsp — APPROVED** [review]

**Review by peternhale — APPROVED** [review]

**Review by RodEsp — DISMISSED** [review]

**Review by RodEsp — APPROVED** [review]

**Review by peternhale — APPROVED** [review]

**Pull request merged**